src/ToolBox/SOS/Strike/disasm.cpp

```diff
@@ -530,7 +530,7 @@ INT_PTR ParseHexNumber (__in_z char *ptr, ___out char **endptr)
         endptr1 = endptr2;
     }
     // if the hex number was specified as 000006fbf9b70f50, an overflow occurred
-    else if (ULONG_MAX == value1 && errno == ERANGE)
+    else if (ULONG_MAX == (ULONG_PTR)value1 && errno == ERANGE)
```

Comment:
The strtoul returns 32 bit unsigned value (long on Windows is always 32 bit). We assign it to INT_PTR which is effectively int32_t on 32 bit platforms and int64_t on 64 bit ones. That means that if it returned ULONG_MAX, the value1 would be "-1" on 32 bit platforms and "0xFFFFFFFF" on 64 bit platforms. So it seems the correct fix is just to cast ULONG_MAX to INT_PTR, that means:
```C++
else if ((INT_PTR)ULONG_MAX == value1 && errno == ERANGE)
```

Reply:
Hmm, I am sorry, I've missed the fact that this code path is 64 bit only. But nevertheless, the change I've suggested still holds.

Reply:
OK, let's see the test results on this. I don't have a preference on this one. I'll go to where the consensus is.